bin/keeper/src/taker.rs

```diff
@@ -157,32 +157,42 @@ pub async fn loop_blocking_orders(
         .unwrap();
     log::info!("Cancelled orders - {:?} for {}", orders, market_name);
 
+    let market_index = mango_client.context.serum3_market_index(&market_name);
+    let s3 = mango_client.context.serum3(market_index);
+
+    let base_decimals = mango_client
+        .context
+        .token(s3.market.base_token_index)
+        .decimals as i32;
+    let quote_decimals = mango_client
+        .context
+        .token(s3.market.quote_token_index)
+        .decimals as i32;
+
     loop {
         interval.tick().await;
 
         let client = mango_client.clone();
         let market_name = market_name.clone();
         let price = price.clone();
 
-        let res = (|| async move {
+        let res: anyhow::Result<()> = (|| async move {
             client.serum3_settle_funds(&market_name).await?;
 
-            let fresh_price = match price.read() {
-                Ok(price) => *price,
-                Err(_) => {
-                    anyhow::bail!("Price RwLock PoisonError!");
-                }
-            };
+            let fresh_price = price.read().unwrap().to_num::<f64>();
+            let bid_price = fresh_price * 1.1;
 
-            let fresh_price = fresh_price.to_num::<f64>();
+            let bid_price_lots =
+                bid_price * 10f64.powi(quote_decimals - base_decimals) * s3.coin_lot_size as f64
+                    / s3.pc_lot_size as f64;
```

Comment:
might be nice to have price_ui_to_serum_lots() or something (I'm unclear on whether `fresh_price` is the ui price or native price or something else)

Reply:
openbook v2 sdk is going to have that imo